libs/fgdbg/src/Session.cpp

```diff
@@ -16,25 +16,39 @@
 
 #include "fgdbg/Session.h"
 #include <sstream>
+#include <set>
+#include <iostream>
 
 namespace filament::fgdbg {
 
 void Session::addPasses(const std::vector<Pass>& passes) {
-    this->passes.insert(this->passes.end(), passes.begin(), passes.end());
+    for (auto& pass: passes) { addPass(pass); }
 }
 
 void Session::addPass(const Pass& pass) {
-    this->passes.push_back(pass);
+    this->passes.insert(std::pair(pass.id, pass));
```

Comment:
instead of the `std::pair(a, b)` syntax in a lot of places you could just use `{a, b}` which is less verbose.